lib/metrics/events.js

```diff
@@ -65,9 +93,87 @@ module.exports = log => {
           return
         }
 
-        return log.flowEvent(event, request)
+        return emitFlowEvent(event, request)
       })
+    },
+
+    /**
+     * Asynchronously emit a flow event indicating the route response.
+     *
+     * @name emitRouteFlowEvent
+     * @this request
+     * @param {Object} response
+     * @returns {Promise}
+     */
+    emitRouteFlowEvent (response) {
+      const request = this
+      const path = request.path.replace(PATH_PREFIX, '')
+
+      if (! FLOW_EVENT_ROUTES.has(path)) {
+        return P.resolve()
+      }
+
+      let status = response.statusCode || response.output.statusCode
+      if (status >= 400) {
+        status = `${status}.${response.errno || 999}`
+      }
+
+      return emitFlowEvent(`route.${path}.${status}`, request)
     }
   }
+
```

Comment:
I absolutely love having this in the metrics module instead of the log module. It feels like the proper place for sure.